langserver/handle_text_document_symbol.go

```diff
@@ -0,0 +1,76 @@
+package langserver
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/haya14busa/go-vimlparser/ast"
+
+	"github.com/sourcegraph/jsonrpc2"
+)
+
+func (h *LangHandler) handleTextDocumentSymbols(ctx context.Context, conn *jsonrpc2.Conn, req *jsonrpc2.Request) (result interface{}, err error) {
+	if req.Params == nil {
+		return nil, &jsonrpc2.Error{Code: jsonrpc2.CodeInvalidParams}
+	}
+
+	var params DocumentSymbolParams
+	if err := json.Unmarshal(*req.Params, &params); err != nil {
+		return nil, err
+	}
+
+	if f, ok := h.files[params.TextDocument.URI]; ok {
+		node, err := f.GetAst()
+		if err != nil {
+			return nil, err
+		} else {
+			return getDocumentSymbols(params, node), nil
+		}
+		return nil, nil
+	} else {
+		return nil, errors.New(fmt.Sprintf("% not open", params.TextDocument.URI))
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
should replace errors.New(fmt.Sprintf(...)) with fmt.Errorf(...)